build-tools/src/main/java/org/elasticsearch/gradle/test/YamlRestTestPlugin.java

```diff
@@ -81,9 +84,13 @@ public void apply(Project project) {
         var cluster = testClusters.register(YAML_REST_TEST);
         TaskProvider<StandaloneRestIntegTestTask> yamlRestTestTask = setupTestTask(project, testSourceSet, cluster);
         project.getPlugins().withType(PluginBuildPlugin.class, p -> {
-            TaskProvider<Zip> bundle = project.getTasks().withType(Zip.class).named(BUNDLE_PLUGIN_TASK_NAME);
-            cluster.configure(c -> c.plugin(bundle.flatMap(Zip::getArchiveFile)));
-            yamlRestTestTask.configure(t -> t.dependsOn(bundle));
+            if (GradleUtils.isModuleProject(project.getPath())) {
```

Comment:
Wouldn't it be better to just use project references here and then have the test cluster convert that to a proper artifact dependency?

Reply:
you mean project reference to itself? that is not quite idiomatic I think. Within a project usually tasks are the way to go I think. consuming configurations are usually only used for sharing artifacts among projects. Haven't seen it been used otherwise anywhere